Source/ProgressionSystemRuntime/Public/Data/PSWorldSubsystem.h

```diff
@@ -14,7 +14,7 @@ enum class ECurrentGameState : uint8;
  * Implements the world subsystem to access different components in the module 
  */
 UCLASS(BlueprintType, Blueprintable, Config = "ProgressionSystem", DefaultConfig)
-class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem
+class PROGRESSIONSYSTEMRUNTIME_API UPSWorldSubsystem : public UWorldSubsystem, public IGameFeatureStateChangeObserver
```

Comment:
Per our conversation:
- remove Observer interface
- remove Observer's virtual methods: `OnGameFeatureLoading` and `OnGameFeatureUnloading`.
-  call manually `UPSWorldSubsystem::Get().OnWorldSubSystemInitialize()` from any PS component's Begin Play method
- call manually `UPSWorldSubsystem::Get().PerformCleanUp()` from any PS component's Destroy method

Reply:
done: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/b6c20c974413272b57157db688195433681af731

Reply:
@h4rdmol, Observer is still present, remove `, public IGameFeatureStateChangeObserver` and its include above

Reply:
removed with: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/09bfccd83e624edaaea9d505ebdd5e754500b050